linkerd/meshtls/rustls/src/client.rs

```diff
@@ -72,15 +72,23 @@ impl Connect {
             .expect("identity must be a valid DNS name");
 
         Self {
+            server_id: client_tls.server_id.into(),
             server_name,
             config,
         }
     }
 }
 
+fn extract_cert(c: &rustls::ClientConnection) -> io::Result<&rustls::Certificate> {
+    match c.peer_certificates().and_then(|certs| certs.first()) {
+        Some(leaf_cert) => io::Result::Ok(leaf_cert),
+        None => Err(io::Error::new(io::ErrorKind::Other, "missing tls end cert")),
+    }
```

Comment:
tiny nit: isn't this match just [`Option::ok_or_else`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.ok_or_else)?